validator_test.go

```diff
@@ -0,0 +1,273 @@
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: AlwaysValid,
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: nil,
+		},
+		{
+			description: "Empty map of Validators success",
+			value: Test{
+				m:                 map[MessageType]Validator{},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: nil,
+		},
+		{
+			description: "Omit default Validators success",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{AlwaysValid},
+				},
+			},
+			expectedErr: nil,
+		},
+		// Failure case
+		{
+			description: "Nil list of default Validators error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: AlwaysValid,
+				},
+				defaultValidators: Validators{nil},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Empty list of Validators error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{},
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Nil Validator error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: nil,
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Nil list of Validators error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{nil},
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Nil map of Validators error",
+			value: Test{
+				m:                 nil,
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.description, func(t *testing.T) {
+			assert := assert.New(t)
+			msgv, err := NewTypeValidator(tc.value.m, tc.value.defaultValidators...)
+			assert.NotNil(msgv)
+			if tc.expectedErr != nil {
+				assert.ErrorIs(err, tc.expectedErr)
+				return
+			}
+
+			assert.NoError(err)
+		})
+	}
+}
+
+func testAlwaysValid(t *testing.T) {
+	assert := assert.New(t)
+	msg := Message{}
+	err := AlwaysValid(msg)
+	assert.NoError(err)
+}
+
+func testAlwaysInvalid(t *testing.T) {
+	assert := assert.New(t)
+	msg := Message{}
+	err := AlwaysInvalid(msg)
+	assert.ErrorIs(err, ErrInvalidMsgType)
+}
+
+func TestHelperValidators(t *testing.T) {
+	tests := []struct {
+		description string
+		test        func(*testing.T)
+	}{
+		{"AlwaysInvalid", testAlwaysInvalid},
+		{"AlwaysValid", testAlwaysValid},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.description, tc.test)
+	}
+}
+
+func TestTypeValidator(t *testing.T) {
+	tests := []struct {
+		description string
+		test        func(*testing.T)
+	}{
+		{"TypeValidator validate", testTypeValidatorValidate},
```

Comment:
Given that output will be `TestTypeValidator/TypeValidator validate`, I think you can remove `TypeValidator` from both of these strings and instead just call them `Validate` and `Factory`.